src/main/java/org/folio/circulation/support/json/JsonArrayPropertyFetcher.java

```diff
@@ -0,0 +1,34 @@
+package org.folio.circulation.support.json;
+
+import static java.util.stream.Stream.empty;
+import static org.folio.circulation.support.json.JsonPropertyFetcher.getArrayProperty;
+
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class JsonArrayPropertyFetcher <T> {
+  private final Function<Object, T> entryMapper;
+
+  JsonArrayPropertyFetcher(Function<Object, T> entryMapper) {
+    this.entryMapper = entryMapper;
+  }
+
+  public Stream<T> toStream(JsonArray array) {
+    if (array == null) {
+      return empty();
```

Comment:
This toStream function is intended for re-use and should have 100 % test coverage.

Reply:
Fully agree, unit tests will be added.